plugins/module_utils/foreman_helper.py

```diff
@@ -1592,6 +1592,39 @@ def run(self, **kwargs):
 
         self.exit_json(task=task)
 
+class KatelloContentImportBaseModule(KatelloAnsibleModule):
+
+    def __init__(self, **kwargs):
+        foreman_spec = dict(
+            path=dict(required=True, type='str'),
+            metadata_file=dict(required=False, type='str'),
+            metadata=dict(required=False, type='dict')
+        )
+        argument_spec = {}
+        foreman_spec.update(kwargs.pop('foreman_spec', {}))
+        argument_spec.update(kwargs.pop('argument_spec', {}))
+
+        self.import_action = kwargs.pop('import_action')
+
+        super(KatelloContentImportBaseModule, self).__init__(foreman_spec=foreman_spec, argument_spec=argument_spec, **kwargs)
+
+        # needs to happen after super().__init__()
+        self.task_timeout = 12 * 60 * 60
+
+    def run(self, **kwargs):
+        metadata_file = self.params.get('metadata_file')
+        self.auto_lookup_entities()
+        payload = _flatten_entity(self.foreman_params, self.foreman_spec)
+
+        if payload.get("metadata") is None and metadata_file:
+            payload["metadata"] = json.load(open(metadata_file))
+            payload.pop("metadata_file")
+
+        if payload.get("metadata") is None:
+            self.fail_json(msg="Metadata not specified.")
```

Comment:
if you pass
```
required_one_of=[['metadata', 'metadata_file']]
```
to the constructor in line 1609, you will get the "metadata not specified" error "for free"